### PR TITLE
Mes 628 cat be legal requirements

### PR DESCRIPTION
**Pull request by gittins7**
## Description

- Added buttons to check legal requirements of 2 normal stops, 1 controlled stop / angled start, uphill start & downhill start to cat B+E Test report page
- Added method to dynamically apply CSS to Normal Start buttons in order to center the text.

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshot - iPad Device
iPad-Screenshot

### CODE REVIEW

**Review by ammarhaiderbjss — APPROVED** [review]
Nice!

File: src/modules/tests/test-data/test-data.constants.ts [hunk 1]
```diff
@@ -57,6 +57,9 @@ export enum LegalRequirements {
   normalStart2 = 'normalStart2',
   angledStart = 'angledStart',
   hillStart = 'hillStart',
+  angledStartControlledStop = 'angledStartControlledStop',
```
Comment (szabikr) [inline src/modules/tests/test-data/test-data.constants.ts:60]: I think we should have category specific LegalRequirements enum

File: src/pages/test-report/components/legal-requirement/legal-requirement.constants.ts [hunk 2]
```diff
@@ -4,4 +4,7 @@ export enum legalRequirementLabels  {
     normalStart2 = 'NS',
     angledStart = 'AS',
     hillStart = 'HS / DS',
+    angledStartControlledStop = 'AS/CS',
```
Comment (szabikr) [inline src/pages/test-report/components/legal-requirement/legal-requirement.constants.ts:7]: Should be a category specific `legalRequirementsLabels`

**Review by szabikr — COMMENTED** [review]
Looks good, I just think that some components should be category specific at this stage.

**Review by szabikr — APPROVED** [review]
Nice one!

File: src/modules/tests/test-data/test-data.selector.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const hasEyesightTestBeenCompleted = (data: TestData) => data.eyesightTes
 export const hasEyesightTestGotSeriousFault = (data: CatBUniqueTypes.TestData) => data.eyesightTest.seriousFault;
 
 export const hasLegalRequirementBeenCompleted = (
-  data: CatBUniqueTypes.TestRequirements, legalRequirement: LegalRequirements) => {
+  data: CatBUniqueTypes.TestRequirements, legalRequirement: LegalRequirements | CatBeLegalRequirements) => {
```
Comment (ammarhaiderbjss) [inline src/modules/tests/test-data/test-data.selector.ts]: eeeek I think I missed the boat on this one, but I don't think separate, category specific enums/labels was a good idea here, having this union type defeats the whole point! The purpose of the Enum is to ensure that the legal requirement being operated upon is a valid legal requirement. The schema is what ensures that the legal requirement is valid for a specific category.
I think we should get revert this unless I've missed a compelling reason to keep it, sorry @gittins7!

Comment (szabikr) [reply src/modules/tests/test-data/test-data.selector.ts]: So this function should deal with only cat-b specific legal requirements

Comment (ammarhaiderbjss) [reply src/modules/tests/test-data/test-data.selector.ts]: ah ok that makes sense, so we create a new `src/modules/tests/test-data/test-data.selector.cat-be.ts` file that exposes the same functions but are typed to `CatBEUniqueTypes`..
We can still just have one set of constants including the enums

File: src/pages/test-report/components/legal-requirement/__tests__/legal-requirement.spec.ts [hunk 4]
```diff
@@ -10,11 +10,11 @@ import { StoreModel } from '../../../../../shared/models/store.model';
 import {
   ToggleLegalRequirement,
 } from '../../../../../modules/tests/test-data/test-requirements/test-requirements.actions';
-import { LegalRequirements } from '../../../../../modules/tests/test-data/test-data.constants';
+import { LegalRequirements, CatBeLegalRequirements } from '../../../../../modules/tests/test-data/test-data.constants';
 import { NavigationStateProvider } from '../../../../../providers/navigation-state/navigation-state';
 import { NavigationStateProviderMock } from '../../../../../providers/navigation-state/__mocks__/navigation-state.mock';
 
-describe('LegalRequirementComponent', () => {
+fdescribe('LegalRequirementComponent', () => {
```
Comment (rstepnowski) [inline src/pages/test-report/components/legal-requirement/__tests__/legal-requirement.spec.ts]: What's the reason behind `fdescribe` here?  Have you forgotten to remove it after testing?

**Review by rstepnowski — APPROVED** [review]
👍

**Pull request merged**